src/test/deploy/functions/release/fabricator.spec.ts

```diff
@@ -0,0 +1,972 @@
+import * as gcfNSV2 from "../../../../gcp/cloudfunctionsv2";
+import * as gcfNS from "../../../../gcp/cloudfunctions";
+import * as pollerNS from "../../../../operation-poller";
+import * as pubsubNS from "../../../../gcp/pubsub";
+import * as schedulerNS from "../../../../gcp/cloudscheduler";
+import * as runNS from "../../../../gcp/run";
+import * as backend from "../../../../deploy/functions/backend";
+import * as scraper from "../../../../deploy/functions/release/sourceTokenScraper";
+import * as planner from "../../../../deploy/functions/release/planner";
+
+describe("Fabricator", () => {
+  // Stub all GCP APIs to make sure this test is hermetic
+  let gcf: sinon.SinonStubbedInstance<typeof gcfNS>;
+  let gcfv2: sinon.SinonStubbedInstance<typeof gcfNSV2>;
+  let poller: sinon.SinonStubbedInstance<typeof pollerNS>;
+  let pubsub: sinon.SinonStubbedInstance<typeof pubsubNS>;
+  let scheduler: sinon.SinonStubbedInstance<typeof schedulerNS>;
+  let run: sinon.SinonStubbedInstance<typeof runNS>;
+
+  beforeEach(() => {
+    gcf = sinon.stub(gcfNS);
+    gcfv2 = sinon.stub(gcfNSV2);
+    poller = sinon.stub(pollerNS);
+    pubsub = sinon.stub(pubsubNS);
+    scheduler = sinon.stub(schedulerNS);
+    run = sinon.stub(runNS);
+
+    gcf.functionFromEndpoint.restore();
+    gcfv2.functionFromEndpoint.restore();
+    scheduler.jobFromEndpoint.restore();
+    gcf.createFunction.rejects(new Error("unexpected gcf.createFunction"));
+    gcf.updateFunction.rejects(new Error("unexpected gcf.updateFunction"));
+    gcf.deleteFunction.rejects(new Error("unexpected gcf.deleteFunction"));
+    gcf.getIamPolicy.rejects(new Error("unexpected gcf.getIamPolicy"));
+    gcf.setIamPolicy.rejects(new Error("unexpected gcf.setIamPolicy"));
+    gcf.setInvokerCreate.rejects(new Error("unexpected gcf.setInvokerCreate"));
+    gcf.setInvokerUpdate.rejects(new Error("unexpected gcf.setInvokerUpdate"));
+    gcfv2.createFunction.rejects(new Error("unexpected gcfv2.createFunction"));
+    gcfv2.updateFunction.rejects(new Error("unexpected gcfv2.updateFunction"));
+    gcfv2.deleteFunction.rejects(new Error("unexpected gcfv2.deleteFunction"));
+    run.getIamPolicy.rejects(new Error("unexpected run.getIamPolicy"));
+    run.setIamPolicy.rejects(new Error("unexpected run.setIamPolicy"));
+    run.setInvokerCreate.rejects(new Error("unexpected run.setInvokerCreate"));
+    run.setInvokerUpdate.rejects(new Error("unexpected run.setInvokerUpdate"));
+    run.replaceService.rejects(new Error("unexpected run.replaceService"));
+    poller.pollOperation.rejects(new Error("unexpected poller.pollOperation"));
+    pubsub.createTopic.rejects(new Error("unexpected pubsub.createTopic"));
+    pubsub.deleteTopic.rejects(new Error("unexpected pubsub.deleteTopic"));
+    scheduler.createOrReplaceJob.rejects(new Error("unexpected scheduler.createOrReplaceJob"));
+    scheduler.deleteJob.rejects(new Error("unexpected scheduler.deleteJob"));
```

Comment:
holy moly

Reply:
Admittedly a bit of a PITA, but not as much of a PITA as you'd expect  with VIM bindings:

1. Use code complete to make a line for each export
2. Use Shift + V to enter visual mode and highlight all lines
3. Use a find & replace: `%s/ +(.*)/\1.rejects(new Error("unexpected \1"));`